crates/store/src/db/mod.rs

```diff
@@ -440,6 +426,78 @@ impl Db {
             .await
     }
 
+    /// Reconstructs account storage at a specific block from the database
+    ///
+    /// This method queries the decomposed storage tables and reconstructs the full
+    /// `AccountStorage` with SMT backing for Map slots.
+    // TODO split querying the header from the content
+    #[instrument(level = "debug", target = COMPONENT, skip_all, ret(level = "debug"), err)]
+    pub async fn select_account_storage_at_block(
+        &self,
+        account_id: AccountId,
+        block_num: BlockNumber,
+    ) -> Result<miden_objects::account::AccountStorage> {
+        self.transact("Get account storage at block", move |conn| {
+            queries::select_account_storage_at_block(conn, account_id, block_num)
+        })
+        .await
+    }
+
+    /// Gets the latest account storage from the database
+    ///
+    /// Uses the `is_latest` flag for efficient querying.
+    #[instrument(level = "debug", target = COMPONENT, skip_all, ret(level = "debug"), err)]
+    pub async fn select_latest_account_storage(
+        &self,
+        account_id: AccountId,
+    ) -> Result<miden_objects::account::AccountStorage> {
```

Comment:
Similar nit as in one of the other comments - we could probably change `<miden_objects::account::AccountStorage` to just `AccountStorage`.

Also, this is also a temporary method, right?